src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -375,6 +375,9 @@ private static void AddSupportedResponseTypes(
                     apiResponseType.ApiResponseFormats.Add(defaultResponseFormat);
                 }
 
+                // We set the Description to the first non-null value we find that matches the status code.
+                apiResponseType.Description ??= responseMetadataTypes.FirstOrDefault(x => x.StatusCode == apiResponseType.StatusCode && x.Description is not null)?.Description;
```

Comment:
Is there a reason the existing logic in the ApiResponseTypeProvider doesn't handle this? That feels like the right place to do this instead of having to filter on the types returned by it to set the description.

Reply:
@sander1095 Thanks for digging into this and sharing detailed notes. Based on this analysis, I expect this option:

> I could change my code to use the LAST description it encounters in the list, in case multiple descriptions are set

To be the prefered one since it aligns most closely with the way metadata gets respected in other scenarios. We'll have to sanity check this implementation with multiple `ProducesResponseType` attributes on the same endpoint to make sure the additions only apply to the target type and status code.

Reply:
@captainsafia Done! See the latest commits